src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/UnmanagedToManagedStubGenerator.cs

```diff
@@ -110,6 +109,9 @@ public BlockSyntax GenerateStubBody(ExpressionSyntax methodToInvoke)
                 allStatements.AddRange(tryStatements);
             }
 
+            allStatements.AddRange(statements.AssignOut);
+            allStatements.AddRange(statements.Cleanup);
```

Comment:
This order concerns me. We added the finally staements above after we added the cleanup. Inverting this seems like a profound change.